### PR TITLE
DOCS DFS cycle detection

### PR DESCRIPTION
**Pull request by Hromz**
- Short description
- Potential uses
- Limitation
- Syntax
- Optional

### CODE REVIEW

**Comment by bobluppes** [comment]
closes #60

File: docs/docs/algorithms/cycle-detection/dfs-based.md [hunk 1]
```diff
@@ -2,4 +2,55 @@
 sidebar_position: 1
 ---
 
-# DFS Based Cycle Detection
\ No newline at end of file
+# DFS Based Cycle Detection
+A DFS based cycle detection algorithm is used to identify cycles in graphs, both directed and undirected.
+The algorithm doesn't work on edges with weight and therefore cannot be used for negative weight cycles.
+
```
Comment (bobluppes) [inline docs/docs/algorithms/cycle-detection/dfs-based.md:8]: [minor]: It sounds like this is a disadvantage of the algorithm, but I am wondering if negative weight cycles would ever influence cycle detection. I would argue the algorithm does work for weighted graphs, but the edge weight will simply not be considered 🙈 

What do you think about this?

```suggestion
The algorithm can be used to detect cycles in the structure of a graph, as it does not consider edge weights.

```

**Review by bobluppes — APPROVED** [review]
Looks very good!
Only one question on which I would value your opinion. Looks good to merge otherwise 👍🏻

File: docs/docs/algorithms/cycle-detection/dfs-based.md (hunk 1 above)
Comment (Hromz) [reply docs/docs/algorithms/cycle-detection/dfs-based.md:8]: I agree that your suggestion is more correct and a bit more readable.

**Review by bobluppes — APPROVED** [review]
Thanks, LGTM

**Pull request merged**